app/Domain/Tickets/Services/Tickets.php

```diff
@@ -2059,6 +2061,157 @@ public function prepareTicketDates(&$values)
 
             return $values;
         }
-    }
 
+        public function pollForNewAccountMilestones(): array | false
+        {
+            return $this->ticketRepository->getAllBySearchCriteria(
+                ["type" => "milestone"],
+                'date'
+            );
+        }
+
+        // since the date attribute of milestones gets updated when the milestone is updated we need to poll for updated milestones
+        // using that date attribute
+        public function pollForUpdatedAccountMilestones(): array|false
+        {
+            $milestones = $this->ticketRepository->getAllBySearchCriteria(
+                ["type" => "milestone"],
+                'date'
+            );
+
+            foreach ($milestones as $key => $milestone) {
+                $milestones[$key]['id'] = $milestone['id'] . '-' . $milestone['date'];
+            }
+
+            return $milestones;
+        }
+
+        // TODO: update this to be todo and update zapier as well 
+        public function pollForNewAccountTickets(): array|false
+        {
+            return $this->ticketRepository->getAllBySearchCriteria(
+                ["excludeType" => "milestone"],
+                'date'
+            );
+        }
+
+        // Since the date attribute of todos gets updated when the todo is updated we need to poll for updated todos
+        // using that date attribute
+        public function pollForUpdatedAccountTodos(): array|false
+        {
+            $todos = $this->ticketRepository->getAllBySearchCriteria(
+                ["excludeType" => "milestone"],
+                'date'
+            );
+
+            foreach ($todos as $key => $todo) {
+                $todos[$key]['id'] = $todo['id'] . '-' . $todo['date'];
+            }
+
+            return $todos;
+        }
+
+        public function createTodoUsingZapier($values): array|bool
```

Comment:
Why this wrapper, it seems the addTicket method would be able to handle this?

Reply:
yep lets use `addTicket` I agree